modules/vertx-swagger-router/src/main/java/io/vertx/ext/swagger/router/SwaggerRouter.java

```diff
@@ -100,9 +101,23 @@ private static void configureRoute(Route route, String serviceId, Operation oper
 
     private static String convertParametersToVertx(String path) {
         Matcher pathMatcher = PATH_PARAMETERS.matcher(path);
+
+        while (pathMatcher.find()) {
+            checkParameterName(pathMatcher.group());
+        }
+
         return pathMatcher.replaceAll(":$1");
     }
 
+    private static void checkParameterName(String parameterPlaceholder) {
+        final Matcher matcher = PATH_PARAMETER_NAME.matcher(parameterPlaceholder);
+
+        if (!matcher.matches()) {
+            final String parameterName = parameterPlaceholder.substring(1, parameterPlaceholder.length() - 1);
+            throw new IllegalArgumentException("Illegal path parameter name: " + parameterName);
```

Comment:
Maybe we should remind the accepted pattern ine the error message ?
Something like : `"Illegal path parameter name : "+ parameterName+". It must match [\p{Alpha}\d_]+"`